packages/mobile/App/services/sync/utils/buildFromSyncRecord.ts

```diff
@@ -42,3 +42,22 @@ export const buildFromSyncRecord = (
   const dbRecord = mapFields(fieldMapping, pick(data, includedColumns));
   return dbRecord;
 };
+
+export const buildForRawInsertFromSyncRecord = (
+  model: typeof BaseModel,
+  { data, isDeleted }: SyncRecord,
+): DataToPersist => {
+  const includedColumns = extractIncludedColumns(model);
+  const record = pick(data, includedColumns);
+  record.isDeleted = isDeleted;
```

Comment:
I feel a bit funny about mixing a column that isn't going into the actual record inside the record object.

Reply:
Oh yeah, I checked the call site and see why you did it. But the weirdness of adding that isDeleted to the record, although it's fine to maintain it, should be restricted to that place, I reckon, rather than in this more generic feeling util. It does get used there, but in a way where it's very specific to how those bulk inserts are handled.